core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/ast/ASTServiceGraphPattern.java

```diff
@@ -84,6 +81,24 @@ public String getPatternString() {
 		return patternString;
 	}
 
+	private String getSourceString() {
+		Node theParent = getParentContainer(this);
+		String sourceString = null;
+		if (theParent instanceof ASTOperationContainer)
+			sourceString = ((ASTOperationContainer) theParent).getSourceString();
```

Comment:
Editorial: could you add open and close brackets here and in other places where you have if/else constructions?

Reply:
done